### DISCUSSION TITLE
Show selected options dynamically for each select element from model in a multiple select

### DISCUSSION THREAD

**Discussion by HermanCeaser** [livewire/livewire#4846] [1. Help]
Hi everyone, Hope you are fine.

I have a table that has a column of roles that should be a multiple select for each of the users.

My question is: how do i show the selected roles for each user in a multiple select.

Below is my blade component.
```
@forelse($staffs as $key => $staff)
  <tr>
    <td>{{ $staff->employee_id }}</td>
    <td>{{ $staff->full_name }}</td>
    <td>{{ $staff->department->department_name }}</td>
    <td style="width:40%">
      <div>
        <select disabled class="form-control select-roles" multiple>
          <option value="" disabled>@lang('messages.select')</option>
          @foreach ($roles as $role)
            <option value="{{ $role->id }}">{{ $role->name }}</option>
          @endforeach
        </select>
      </div>
    </td>
@empty
  <tr>
    <td valign="top" colspan="6">No data available<br>
      <span class="text-success bolds">
        <i class="fa fa-arrow-left"></i>
        Add new records.
      </span>
    </td>
  </tr>
@endforelse


```

My Livewire component is as follows.

```
    public $school;
    public $staffs = [];
   public $roles = [];
    // public $selected_roles = [1, 4];
    public $editRoleIndex = -1;


    public function mount()
    {
        $this->roles = Role::all();
        $this->staffs = Staff::with('user', 'department')->where('tenantID', $this->school->tenantID)->get();
    }

   public function render()
   {
        return view('livewire.import-users-into-school');
   }



```

- I know I am supposed to bind the select to some public variable let\'s say `public $selected_roles = []` and a `wire:model="selected_roles"` on my select but  the problem is that my select is dynamic and the options are from some model.  
- Here is what i my end result would look like?
![roles_peermissions](https://user-images.githubusercontent.com/41926048/160806664-a85683d9-a807-4ac7-a87e-017e57f4f535.png)

### Question
1. How do i get to see which role is attached to which user?
2. What should i bind to my select element such that livewire can populate the selected roles automatically and dynamically?

Any help is welcome.

**Comment by joshhanley** [comment]
@HermanCeaser you would have to ensure you bind to a property which contains the user's selected role data. Like `wire:model="selectedRoles.{{ $staff->id }}"`.

On mount you would need to get each user's selected roles and populate them into the `selectedRoles` property in the format like above.

Hope this helps!

**Reply by HermanCeaser** [reply]
Yeah It does help @joshhanley , glad i had figured a way to do that, i had to get each users role from the staff collection using array_map in the mount method

``` 
$this->selected_roles = $this->staffs->map(
     function($staff, $key){ 
         return $staff->user->roles->pluck('name')->toArray(); 
      }
);

```

Then in the view, I had to bind like `<select wire:model="selected_roles.{{$key}}" class="form-control select-roles" multiple>`

And it worked

**Comment by HermanCeaser** [comment]
How would i update the changed selected_role?  My knowledge is that i have to write an updatedSelectedRole function but the selectedRole is dynamic.
